src/mock_server/bare_server.rs

```diff
@@ -190,6 +194,35 @@ impl MockGuard {
         let (mounted_mock, _) = &state.mock_set[self.mock_id];
         mounted_mock.received_requests()
     }
+
+    pub async fn satisfied(&self) {
+        let MockGuard {
+            mock_id,
+            server_state,
+            notify,
+        } = self;
+        let notification = notify.0.notified();
+        tokio::pin!(notification); // std::pin::pin was added in 1.68.0
+        if notification.as_mut().enable() {
+            // reraise a signal just in case
+            notify.0.notify_waiters();
+            return;
+        }
+
+        if self.notify.1.load(std::sync::atomic::Ordering::Acquire) {
+            return;
+        }
+
+        let state = server_state.read().await;
+        let report = state.mock_set.verify(*mock_id);
+        if report.is_satisfied() {
+            // reraise a signal just in case another waiter joined the queue
+            notify.0.notify_waiters();
+            return;
+        }
```

Comment:
Why do need an explicit verification check here? What kind of sequence of events would make this necessary to avoid missing a notification?